android/src/main/java/com/clevertap/react/CleverTapModuleImpl.java

```diff
@@ -1419,12 +1192,30 @@ public void onInAppsFetched(final boolean isSuccess) {
                 }
             });
         } else {
-            String error = ErrorMessages.CLEVERTAP_NOT_INITIALIZED.getErrorMessage();
+            String error = ErrorMessages.CLEVERTAP_NOT_INITIALIZED;
             callbackWithErrorAndResult(callback, error, null);
         }
     }
 
+    public void onEventListenerAdded(String eventName) {
+        CleverTapEvent event = CleverTapEvent.fromName(eventName);
+        if (event == null) {
+            Log.e(TAG, "Event listener added for unsupported event " + eventName);
+            return;
+        }
+        // disable the buffering for the specified event as it already has attached listener and
+        // flush all buffered events
+        CleverTapEventEmitter.INSTANCE.disableBuffer(event);
+        CleverTapEventEmitter.INSTANCE.flushBuffer(event);
+    }
 
+    private void enableEventEmitter(ReactContext reactContext) {
+        CleverTapEventEmitter.INSTANCE.setReactContext(reactContext);
+        // disable buffers after a delay in order to give some time for event listeners to attach
+        // and receive initially buffered events. After that all buffers will be cleared and disabled
+        // and event will continue to be sent immediately.
+        new Handler().postDelayed(() -> CleverTapEventEmitter.INSTANCE.resetAllBuffers(false), 5000);
+    }
```

Comment:
rather than 5seconds which might be ANR time, we can support additional functionality so we dont have to put such a post delayed and let clients call a method synonymous to `flush` on Component ready, incase they dont we can have this postdelayed solution, lets give them both options.

Also this does not handle the situation where react framework itself recreates component itself due to any reasons (I mean such a thing might be implemented on their end later on as well)

Reply:
We could add a new method for notifying when the client has added all his listeners, but if we want to be backwards compatible we should also have this automatic delayed resetting. The React component being destroyed would be ok if the NativeModule remains alive - I am not sure if this is always the case, but it seems that way to me. Let's have a chat about this, maybe we can think of something better.